internal/venice-common/src/main/java/com/linkedin/venice/meta/Partition.java

```diff
@@ -102,13 +128,23 @@ public List<Instance> getInstancesInState(String state) {
   public List<Instance> getReadyToServeInstances() {
     List<Instance> completed = this.executionStatusToInstancesMap.get(ExecutionStatus.COMPLETED);
     List<Instance> online = this.helixStateToInstancesMap.get(HelixState.ONLINE);
-    return completed.size() > online.size() ? completed : this.readyInstances;
+    return completed.size() > online.size() ? completed : getReadyInstances();
   }
 
   public List<Instance> getWorkingInstances() {
```

Comment:
Also, just to be cautious as I am confused by the existing variable/API naming here. Is working instance containing the ready-to-serve and bootstrapping ones?

Reply:
Yeah... although I think that's a hangover from the O/B/O days, since we don't even have `BOOTSTRAP` anymore... so most likely, `getWorkingInstances()` could be substituted for `getReadyToServeInstances()` and it would work just as well...

Reply:
I ended up cleaning this up. Deeeeeep rabbit hole. But I think with this we'll be completely rid of O/B/O. I even deleted `ONLINE` and `BOOTSTRAP` from the `HelixState` enum!